base/arrayshow.jl

```diff
@@ -480,12 +480,21 @@ function typeinfo_prefix(io::IO, X)
     # what the context already knows about the eltype of X:
     eltype_ctx = typeinfo_eltype(typeinfo)
     eltype_X = eltype(X)
-    # Types hard-coded here are those which are created by default for a given syntax
-    if eltype_X == eltype_ctx || !isempty(X) && eltype_X in (Float64, Int, Char, String)
-        ""
-    elseif print_without_params(eltype_X)
-        string(unwrap_unionall(eltype_X).name) # Print "Array" rather than "Array{T,N}"
+    if X isa AbstractDict
+        K, V = typeof(X).parameters
```

Comment:
This is only guaranteed to work for `Dict` AFAIK. Use `keytype` and  `valuetype` instead.

Reply:
Thanks!